pulp_deb/app/tasks/synchronizing.py

```
@@ -577,19 +586,60 @@ async def _handle_distribution(self, distribution):
         await asyncio.gather(*sub_tasks)
 
     async def _handle_component(
-        self, component, release, release_file, file_references, architectures
+        self,
+        component,
+        release,
+        release_file,
+        file_references,
+        architectures,
+        no_support_for_architecture_all,
     ):
         # Create release_component
         release_component_dc = DeclarativeContent(
             content=ReleaseComponent(component=component, release=release)
         )
         release_component = await self._create_unit(release_component_dc)
+
+        # Try handling the architecture='all' index first (if present). That way, we can recover the
+        # special case, where a partial mirror does not mirror this index inspite of indicating in
+        # the mirrored metadata that it should.
+        if "all" in architectures and "Packages" in no_support_for_architecture_all:
+            architectures.remove("all")
+            try:
+                await asyncio.gather(
+                    self._handle_package_index(
+                        release_file=release_file,
+                        release_component=release_component,
+                        architecture="all",
+                        file_references=file_references,
+                        no_support_for_architecture_all=no_support_for_architecture_all,
```

Wondering if we couldn't pass a boolean here.

---

...or yet to come :see_no_evil: 

Given that the value itself is not a boolean or boolean-like value I am with @quba42 here :smile:

---

I basically went with fail hard and fail early. If somebody tries to sync a Release file containing the `No-Support-for-Architecture-all` with any value other than `Packages` I throw an error instructing the user to open an issue. Such a case probably would signify some crazy new repo format with a thousand edge cases we can't handle anyway, so better to just have them open the issue...

I now have a short bit at the start for "extracting and interpreting" the `No-Support-for-Architecture-all` value, which then sets the internal `hybrid_format` boolean, which is then used throughout the rest of the sync.

---

That sounds like an excellent plan, not trying to cover unknown usecases by accident.